server/utils/scraping.js

```diff
@@ -0,0 +1,122 @@
+const puppeteer = require('puppeteer');
+
+function domainName(url) {
+	const urlBeginning = /^https?:\/\/?w+\./;
+	const ebayRe = /^\bebay\b/;
+	const amazonRe = /^\bamazon\b/;
+	const craigslistRe = /\b\.craigslist\.\b/;
+	let domain = '';
+	let domainNoEnd = url.replace(urlBeginning, '');
+
+	if (ebayRe.test(domainNoEnd)) {
+		domain = 'ebay';
+	} else if (amazonRe.test(domainNoEnd)) {
+		domain = 'amazon';
+	} else if (craigslistRe.test(domainNoEnd)) {
+		domain = 'craigslist';
+	} else {
+		domain = 'Please enter an ebay or amazon domain';
+	}
+	return domain;
+}
+const scraping = async (url) => {
+	let pageData = {};
+
+	try {
+		const browser = await puppeteer.launch({ headless: true });
+		const page = await browser.newPage();
+		await page.goto(url);
+
+		//This makes sure that the page has loaded based on one selector
+		if (domainName(url) === 'amazon') {
+			await page.waitForSelector('span#productTitle');
+
+			pageData = await page.evaluate(() => {
+				const priceBeginning = /[a-zA-Z]*/;
+				const dollarSign = /[$]/;
+				//nothing inside or outside of page.evaluate is available outside of it's closure, because it is running purely in the dom
+				let titleData = document.getElementById(`productTitle`);
+				let priceData = document.getElementById(`price_inside_buybox`);
```

Comment:
If it is easier to scrape the price from `price_inside_buybox`, then no worries. Else you can try scraping from `priceblock_ourprice` too!

Reply:
yeah priceblock_ourprice was actually what I was going by first! But that ID cd changes when an item is on sale to something like "dealprice" etc. I found the buy box was more consistent when applied to the admittedly small sample size of like 10 different amazon pages.